benchmarks/vectorization_strategy_benchmark.py

```diff
@@ -0,0 +1,700 @@
+        )
+        center_y = tf.random.uniform(
+            shape=[],
+            minval=0,
+            maxval=image_height,
+            dtype=tf.int32,
+            seed=self.seed,
+        )
+        return center_x, center_y
+
+    def _compute_rectangle_size(self, inputs):
+        input_shape = tf.shape(inputs)
+        image_height, image_width = (
+            input_shape[0],
+            input_shape[1],
+        )
+        height = tf.random.uniform(
+            [],
+            minval=self.height_lower,
+            maxval=self.height_upper,
+            dtype=tf.float32,
+        )
+        width = tf.random.uniform(
+            [],
+            minval=self.width_lower,
+            maxval=self.width_upper,
+            dtype=tf.float32,
+        )
+
+        if self._height_is_float:
+            height = height * tf.cast(image_height, tf.float32)
+
+        if self._width_is_float:
+            width = width * tf.cast(image_width, tf.float32)
+
+        height = tf.cast(tf.math.ceil(height), tf.int32)
+        width = tf.cast(tf.math.ceil(width), tf.int32)
+
+        height = tf.minimum(height, image_height)
+        width = tf.minimum(width, image_width)
+
+        return height, width
+
+    def _compute_rectangle_fill(self, inputs):
+        input_shape = tf.shape(inputs)
+        if self.fill_mode == "constant":
+            fill_value = tf.fill(input_shape, self.fill_value)
+        else:
+            # gaussian noise
+            fill_value = tf.random.normal(input_shape)
+
+        return fill_value
+
+    def get_config(self):
+        config = {
+            "height_factor": self.height_factor,
+            "width_factor": self.width_factor,
+            "fill_mode": self.fill_mode,
+            "fill_value": self.fill_value,
+            "seed": self.seed,
+        }
+        base_config = super().get_config()
+        return dict(list(base_config.items()) + list(config.items()))
+
+
+"""
+# Benchmarking
+"""
+(x_train, _), _ = keras.datasets.cifar10.load_data()
+x_train = x_train.astype(float)
+
+x_train.shape
+
+
+images = []
+
+num_images = [1000, 2000, 5000, 10000, 25000, 37500, 50000]
+
+results = {}
+
+for aug in [VectorizedRandomCutout, VMapRandomCutout, MapFnRandomCutout]:
+    c = aug.__name__
+    layer = aug(0.2, 0.2)
+    runtimes = []
+    print(f"Timing {c}")
+
+    for n_images in num_images:
+        # warmup
+        layer(x_train[:n_images])
+
+        t0 = time.time()
+        r1 = layer(x_train[:n_images])
+        t1 = time.time()
+        runtimes.append(t1 - t0)
+        print(f"Runtime for {c}, n_images={n_images}: {t1-t0}")
+
+    results[c] = runtimes
+
+plt.figure()
+for key in results:
+    plt.plot(num_images, results[key], label=key)
+    plt.xlabel("Number images")
+
+plt.ylabel("Runtime (seconds)")
+plt.legend()
+plt.show()
+
+"""
+# Sanity check
+all of these should have comparable outputs
+"""
+
+images = []
+for aug in [VectorizedRandomCutout, VMapRandomCutout, MapFnRandomCutout]:
```

Comment:
Would it be useful to also have numbers with XLA compilation?

Reply:
https://github.com/keras-team/keras-cv/issues/165 lets do in a follow up.

Reply:
(but I do think so)

Reply:
In this case for Vectorized and VMap:

![immagine](https://user-images.githubusercontent.com/1710528/157075068-dc13c034-ce1c-4873-bde9-992027ab42dd.png)

Reply:
We also need to consider the general discussion we had at https://github.com/keras-team/keras-cv/pull/146#issuecomment-1049352918

As this one is a special case that is running fine with `vectorized_map/fallback_to_while_loop=False`.

Also,  this simple case doesn't have any specific not JITable ops like e.g. `ImageProjectiveTransformV3` (E.g. for rotation) or  `tf.bincount` (see https://github.com/keras-team/keras-cv/issues/141#issuecomment-1047817015)